presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -14,15 +14,176 @@
 package io.prestosql.plugin.jdbc;
 
 import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import io.airlift.units.Duration;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.SchemaTableName;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
 import java.util.Set;
 
+import static com.google.common.collect.Sets.difference;
 import static io.prestosql.spi.testing.InterfaceTestUtils.assertAllMethodsOverridden;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static org.assertj.core.api.Assertions.assertThat;
 
+@Test(singleThreaded = true)
 public class TestCachingJdbcClient
 {
+    private static final Duration FOREVER = Duration.succinctDuration(1, DAYS);
+    private static final ConnectorSession session = testSessionBuilder().build().toConnectorSession();
+    private static final JdbcIdentity identity = JdbcIdentity.from(session);
+
+    private TestingDatabase database;
+    private CachingJdbcClient cachingJdbcClient;
+    private JdbcClient dbHacker;
+    private String schema;
+
+    @BeforeMethod
+    public void setUp()
+            throws Exception
+    {
+        database = new TestingDatabase();
+        cachingJdbcClient = createCachingJdbcClient(true);
```

Comment:
Please create it every test. I noticed that sometimes it is created and sometimes it is created here. So lets be coherent.

Reply:
You are right, however this is a trade-off. This is the only one `cachingJdbcClient` 'override' of the 'default setup' provided by class fields which are widely used. The same could be said about `schema` which is not used by `testSchemaNamesCached` and that's why it's could be more clear to define in each other test method. As the most exaggerated example `database` is not used by preexisting test methods, and possibly could be included to new methods.
There is no difference for Tests clarity if the default value is overriden in test method or just not used in test method. 
I'll left it as is unless you have better arguments.